synapse/storage/databases/state/store.py

```diff
@@ -228,6 +251,142 @@ def _get_state_for_group_using_cache(
 
         return state_filter.filter_state(state_dict_ids), not missing_types
 
+    def _get_state_for_group_gather_inflight_requests(
+        self, group: int, state_filter: StateFilter
+    ) -> Tuple[Sequence[AbstractObservableDeferred[StateMap[str]]], StateFilter]:
+        """
+        Attempts to gather in-flight requests and re-use them to retrieve state
+        for the given state group, filtered with the given state filter.
+
+        Returns:
+            Tuple of two values:
+                A sequence of ObservableDeferreds to observe
+                A StateFilter representing what else needs to be requested to fulfill the request
+        """
+
+        inflight_requests = self._state_group_inflight_requests.get(group)
+        if inflight_requests is None:
+            # no requests for this group, need to retrieve it all ourselves
+            return (), state_filter
+
+        state_filter_left_over = state_filter
+        reusable_requests = []
+        for (
+            request_state_filter,
+            request_deferred,
+        ) in inflight_requests.items():
```

Comment:
I suspect this can fit on a single line?

```suggestion
        for (request_state_filter, request_deferred) in inflight_requests.items():
```

Reply:
I suspect the identifiers were once longer, then black split the lines, but apparently doesn't bother to recombine later on after the identifiers are changed to something shorter. Thanks :)